openforcefield/typing/chemistry/environment.py

```diff
@@ -80,38 +80,58 @@ def _convert_embedded_SMIRKS(smirks):
     # new_smirks = [#1:1]~[#6]
     """
     a_out = 0
-    while smirks.find('$') != -1:
+    while smirks.find('$(') != -1:
         # Find first atom
         atom, a_in, a_out = _find_embedded_brackets(smirks, '\[', '\]')
-        d = atom.find('$')
+        d = atom.find('$(')
         # Find atom with the $ string embedded
         while d == -1:
             atom, temp_in, temp_out = _find_embedded_brackets(smirks[a_out+1:], '\[', '\]')
             a_in = a_out + temp_in + 1
             a_out += temp_out + 1
-            d = atom.find('$')
+            d = atom.find('$(')
 
         # Store the smirks pattern before and after the relevant atom
         pre_smirks = smirks[:a_in]
         post_smirks = smirks[a_out+1:]
 
+        # Check for ring index, i.e. the 1s in "[#6:1]1-CCCCC1"
```

Comment:
Were we not handling ring subsearches like this before? If not we probably want to note this prominently in listing the changes this PR brings in. (Basically just we want to make it easy to tell where/when we made any especially important/big changes.)